src/Symfony/Component/Form/FormFactory.php

```diff
@@ -81,7 +81,7 @@ public function createBuilder($type = 'Symfony\Component\Form\Extension\Core\Typ
             $typeName = $type;
         }
 
-        if (null === $name) {
+        if (null === $name && is_string($typeName)) {
```

Comment:
It would be better to throw an UnexpectedTypeException above just as in createNamedBuilder. Because otherwise the $name variable stays null, which is also wrong and not accepted by createNamedBuilder.